### PR TITLE
feat: add information necessary to create typescript types

### PR DESCRIPTION
**Pull request by dsherret**
The type information of functions and globals is now included in the output. The parser will continue on if it can't get this information out though, so that's why it stores this information in a `Result`.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -275,10 +545,10 @@ fn parse_export_type(input: &[u8]) -> ParseResult<Export> {
   let (input, index) = parse_var_uint(input)?;
 
   let export_type = match kind_byte {
-    0x00 => ExportType::Function,
+    0x00 => ExportType::Function(Err(ParseError::UnresolvedExportType)),
```
Comment (bartlomieju) [inline src/lib.rs:548]: Just to confirm - this is the default that will be populated later on (if it's possible)

**Review by bartlomieju — APPROVED** [review]
LGTM, this is really cool 👍 I assume the end-goal is that we're gonna use this information to generate fake `.d.ts` files on the fly so that Intelisense knows the proper types?

File: src/lib.rs (hunk 1 above)
Comment (dsherret) [reply src/lib.rs:548]: Yeah, it's populated later or left as-is for `skip_types: true`. Then in the CLI we can mark exports that have this as an `unknown` type until we can come up with a fix.

Comment (bartlomieju) [reply src/lib.rs:548]: Lovely

**Pull request merged**